maptivist/src/main/java/com/google/sps/servlets/VoteServlet.java

```diff
@@ -35,51 +39,53 @@
 import java.util.Set;
 import java.util.List;
 import java.util.Base64;
- 
- 
+
+
 /** Servlet that handles all my received marker data */
 @WebServlet("/votes")
 public final class VoteServlet extends HttpServlet {
     
-    private static String voteCount;
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
  
         Gson gson = new Gson();
  
-        response.setContentType("text/html;");
-        response.getWriter().println(voteCount);
+        response.setContentType("application/json;");
+        response.getWriter().println(gson.toJson(Globals.voteCount));
     }
  
  
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        Boolean voteCheck = Boolean.parseBoolean(request.getParameter("vote-choice"));
-        String title = request.getParameter("title");
+        String titleMatch = request.getParameter("title");
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        updateVotes(datastore, titleMatch);
 
-        
-        voteCount = Integer.toString(updateVotes(datastore, title));
         response.sendRedirect("/index.html");
     }
  
 
-    public Entity getEntity(DatastoreService datastore, String title) {
-        Query query = new Query("Marker");
-        query.addFilter("title", Query.FilterOperator.EQUAL, title); 
+    public Entity getEntity(DatastoreService datastore, String titleMatch) {
+        Query query = new Query("Marker"); 
+        query.addFilter("title", FilterOperator.EQUAL, titleMatch); 
+        PreparedQuery pq = datastore.prepare(query);    
+        Entity entity = pq.asSingleEntity();
 
-        PreparedQuery results = datastore.prepare(query);
-        Entity marker = results.asSingleEntity();
-        return marker;
+        return entity;
     }
 
-    public int updateVotes(DatastoreService datastore, String title) {
-        Entity markerEntity = getEntity(datastore, title);
-        Key markerKey = markerEntity.getKey();
-        int votes = Integer.parseInt((String) markerEntity.getProperty("votes"));
+    public void updateVotes(DatastoreService datastore, String titleMatch) {
+        Entity entity = getEntity(datastore, titleMatch);
+        int votes = (int) (long) entity.getProperty("votes");
+
         votes += 1;
-        markerEntity.setProperty("votes", votes);
-        datastore.put(markerEntity);
-        return votes;
+        Globals.voteCount = Integer.toString(votes);
+        entity.setProperty("votes", votes);
+
+        datastore.put(entity);        
+    }
+
+    static class Globals {
+        public static String voteCount;
```

Comment:
Again, not a fan of using the a global here.

Instead we should be fetching the Marker from the Datastore and returning its votes here.